Add the -no-absname option to ocamlc, ocamlopt and ocamldep

Abiola-Zeenat opened the pull request:
Addressing #5561 
I am adding a -no-absname #counterpart to the -absname flag that matches the current default of the compiler.

Abiola-Zeenat committed:
Add the -no-absname option to ocamlc, ocamlopt and ocamldep

dra27 commented on driver/main_args.ml:
```diff
@@ -28,7 +29,11 @@ let mk_alert f =
 
 let mk_absname f =
-  "-absname", Arg.Unit f, " Show absolute filenames in error messages"
+  "-absname", Arg.Unit f, " Show absolute filenames in error messages "
```
I think it's a typo to have a space at the end of this string?
```suggestion
  "-absname", Arg.Unit f, " Show absolute filenames in error messages"
```

Abiola-Zeenat replied:
Yes, It is indeed a typo, corrections will be made.

dra27 commented on man/ocaml.1:
```diff
@@ -70,6 +70,9 @@ The following command-line options are recognized by
 Show absolute filenames in error messages.
 .TP
+.B \-no-absname
+Do not Show absolute filenames in error messages.
```
Lowercase 's' wanted on show here (and in the other two files):
```suggestion
Do not show absolute filenames in error messages.
```

Octachron replied:
There is also the slight issue that the negation is inexact: if ocamlc is called on a an absolute file name, `ocamlc -no-absname /absolute/path/file.ml` will show this absolute filename without trying to make it relative to the current working directory.
I would propose to make that point more obvious with:
```
Do not normalize filenames to absolute filenames in error messages.
```

Abiola-Zeenat replied:
okay, noted. Corrections will be made.

dra27 reviewed (commented):
Couple of little typos, but this otherwise looks fine to me, thanks!

Octachron commented on driver/main_args.ml:
```diff
@@ -1,3 +1,4 @@
+
```
This is a spurious new line that can be removed.

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/21 03:10 -0700):
> I am adding a -no-absname #counterpart to the -absname flag that
> matches the current default of the compiler.

This is doubly welcome, many thnaks!!

 * Because indeed each boolean option needs a counterpart
 * Because, personnally, I find those absolute paths cumbersome

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/21 09:31 -0700):
> @Abiola-Zeenat commented on this pull request.
> 
> 
> 
> > @@ -70,6 +70,9 @@ The following command-line options are recognized by
>  .B \-absname
>  Show absolute filenames in error messages.
>  .TP
> +.B \-no-absname
> +Do not Show absolute filenames in error messages.
> 
> okay, noted. Corrections will be made.

It's best if you can report once the corrections actually have been
made, rather than reporting before. So we know we can review again.

shindere commented:
Regarding the commit message, it would be good if you could change it.

Suggestion:

Add the -no-absname option to ocamlc and ocamlopt

You can use `git log` to see other commit messages and try to adopt a
similar style. You can then change the commit message with `git commit
--amend`.

Also, the LaTeX manual will need to be updated, too. See
`manual/src/cmds/unified-options.etex`.

And, it seems `ocamldep` also has an `-absname` command-line option so
this tool may benefit from an update, too.

Abiola-Zeenat commented:
> Regarding the commit message, it would be good if you could change it. Suggestion: Add the -no-absname option to ocamlc and ocamlopt You can use `git log` to see other commit messages and try to adopt a similar style. You can then change the commit message with `git commit --amend`. Also, the LaTeX manual will need to be updated, t

Abiola-Zeenat commented:
> Regarding the commit message, it would be good if you could change it. Suggestion: Add the -no-absname option to ocamlc and ocamlopt You can use `git log` to see other commit messages and try to adopt a similar style. You can then change the commit message with `git commit --amend`. Also, the LaTeX manual will need to be updated, too. See `manual/src/cmds/unified-options.etex`. And, it seems `ocamldep` also has an `-absname` command-line option so this tool may benefit from an update, too.

All necessary corrections based on your suggestion has been made, kindly review

shindere commented:
Sorry but your branch looks a bit disorganised with weird merge commits.
Could you please lcean it up and make sure everything is in one commit
with a proper commit message?

Abiola-Zeenat commented:
> Sorry but your branch looks a bit disorganised with weird merge commits. Could you please lcean it up and make sure everything is in one commit with a proper commit message?

Yes, I have done that.

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/24 14:54 -0700):
> > Sorry but your branch looks a bit disorganised with weird merge commits. Could you please lcean it up and make sure everything is in one commit with a proper commit message?
>
> Yes, I have done that.

Many thanks @Abiola-Zeenat, it's much better, indeed!

As a commit message, `Add the -no-absname option to ccamlc and ocamlopt`
would be enough (add a dash in front of the option, and lowercase the
first letter of the names of the compilers). The rest of the message `I
made some changes to the previous commit #11653` should be removed
because the previous commit does not exist any longer.

In `driver/main_args.ml`, in the definition of `mk_no_absname`, the `d`
of `default` should be in lower case.

Also, I find the current help text of the option a bit long. I realise
it's @Octachron proposal, but still, let me propose something more concise:

```
" Do not try to show absolute filenames in error messages (default)"
```

That would avoid the repetition of `filenames`, which I find a bit
cumbersome. If you change this message, you'll need to update the manual
pages, too.

Also, regarding `ocamldep`, you have added the option to its manual
page, which is great, but you did not add it to the tool itself. The
file you need to modify is `driver/makedepend.ml`.

Abiola-Zeenat commented:
> Abiola Zeenat Abdulsalam (2022/10/24 14:54 -0700):
> > Sorry but your branch looks a bit disorganised with weird merge commits. Could you please lcean it up and make sure everything is in one commit with a proper commit message? Yes, I have done that.
> Many thanks @Abiola-Zeenat, it's much better, indeed! As a commit message, `Add the -no-absname option to ccamlc and ocamlopt` would be enough (add a dash in front of the option, and lowercase the first letter of the names of the compilers). The rest of the message `I made some changes to the previous commit #11653` should be removed because the previous commit does not exist any longer. In `driver/main_args.ml`, in the definition of `mk_no_absname`, the `d` of `default` should be in lower case. Also, I find the current help text of the option a bit long. I realise it's @Octachron proposal, but still, let me propose something more concise: ``` " Do not try to show absolute filenames in error messages (default)" ``` That would avoid the repetition of `filenames`, which I find a bit cumbersome. If you change this message, you'll need to update the manual pages, too. Also, regarding `ocamldep`, you have added the option to its manual page, which is great, but you did not add it to the tool itself. The file you need to modify is `driver/makedepend.

@shindere, kindly review the new commit

shindere commented:
Things improve, thanks.

In the ocamldep sources, could you please add a `(default)` annotation
to the option that gives the default behaviour?

You should also add an entry to the `Changes` file and also run

```
tools/check-typo
```
Notice that it's not happy and make it happy.

And, since you will be amending this commit, could you please write the
names of the compilers all in lowercase in the commit message?

Abiola-Zeenat commented:
> Things improve, thanks. In the ocamldep sources, could you please add a `(default)` annotation to the option that gives the default behaviour? You should also add an entry to the `Changes` file and also run ``` tools/check-typo ``` Notice that it's not happy and make it happy. And, since you will be amending this commit, could you please write the names of the compilers all in lowercase in the commit message?

All corrections made.

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/27 10:09 -0700):
> All corrections made.

Thanks.

Ideally, the commit message should reflect what you are doing as
accurately as possible. Now that you also have updated ocamldep, why not
include this in your commit message?

Suggestion: replace `Add the -no-absname option to ocamlc and ocamlopt`
with `Add the -no-absname option to ocamlc, ocamlopt and ocamldep`.

(of couorse the Changes entry) should then be updated, too.)

Also, I have two questions for you, as exercises (if you do the Changes
above I think the PR will be ready to be merged even if you are not able
to respond to these questions, but perhaps try...)

 1. ALthough it's not strictly necessary, would you be able to rebase your branch on latest trunk?
 2. Did you test your modifications? If I asked you how I should do to make sure your modifications work, what would you advise?

And finally, once you will be done with this PR, would you like to implement another, similar one which would introduce the `-no-g` option?

shindere commented:
Also, notice that not all the automatic checks pass.

Can you try to find out why and fix your PR so that all the tests pass?

Abiola-Zeenat commented:
> Abiola Zeenat Abdulsalam (2022/10/27 10:09 -0700):
> All corrections made.
> Thanks. Ideally, the commit message should reflect what you are doing as accurately as possible. Now that you also have updated ocamldep, why not include this in your commit message? Suggestion: replace `Add the -no-absname option to ocamlc and ocamlopt` with `Add the -no-absname option to ocamlc, ocamlopt and ocamldep`. (of couorse the Changes entry) should then be updated, too.) Also, I have two questions for you, as exercises (if you do the Changes above I think the PR will be ready to be merged even if you are not able to respond to these questions, but perhaps try...) 1. ALthough it's not strictly necessary, would you be able to rebase your branch on latest trunk? 2. Did you test your modifications? If I asked you how I should do to make sure your modifications work, what would you advise? And finally, once you will be done with this PR, would you like to implement another, similar one which would introduce the `-no-g` option?

I have updated the commit message as well as the changes file. Also to answer your questions
1. I rebased my branch to the trunk branch and I got a response that my branch is up to date.
2. Yes, I tested my modification, I did that by running `./ocamlopt.opt -I stdlib -absname -no-absname test.ml` on a test.ml file that yields an error, then I checked for the filename in the error message.
The screenshot of the test has been attached.(when absname was called,the absolute filename was shown in the error message whereas it didn't when -no-absname was also added.
3. Yes, I would like to implement another option.
![image](https://user-images.githubusercontent.com/81884414/198564190-498a3d70-a0da-43eb-8fa8-404b0ec2c66f.png)

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/28 03:19 -0700):
> I have updated the commit message as well as the changes file.

I can indeed see these changes. Thanks.

> Also to answer your questions
> 1. I rebased my branch to the trunk branch and I got a response that
> my branch is up to date.

Yes, but that updated only your local copy of the branch. For this
rebase to become visible, youshould push the branch to your public fork
as you do after having amended a commit.

> 2. Yes, I tested my modification, I did that by running
> `./ocamlopt.opt -I stdlib -absname -no-absname test.ml` on a test.ml
> file that yields an error, then I checked for the filename in the
> error message.

Okay I am convinced by your explanations, thanks!

Just so that you know (but you couldn't guess, don't worry): since
screenshots are images they are not readable by visually impaired
persons. So please, as much as possible, provide things like commands,
program outputs etc. in a textual rather than graphical form.

Finally, since you'll have to push your branch again anyway, please
remote the final dot at the end of the commit message, but not in
Changes.

Abiola-Zeenat commented:
> Yes, but that updated only your local copy of the branch. For this rebase to become visible, youshould push the branch to your public fork as you do after having amended a commit.

Okay, I get it now, could you check if it has been updated now

> 2. Yes, I tested my modification, I did that by running `./ocamlopt.opt -I stdlib -absname -no-absname test.ml` on a test.ml file that yields an error, then I checked for the filename in the error message.
> Okay I am convinced by your explanations, thanks! 
Just so that you know (but you couldn't guess, don't worry): since screenshots are images they are not readable by visually impaired persons. So please, as much as possible, provide things like commands, program outputs etc. in a textual rather than graphical form.

Oh! I really didn't guess. I would ensure not to make such mistake again.

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/28 06:40 -0700):
> Okay, I get it now, could you check if it has been updated now

The commit message has indeed been updated, but the rebase is not
visible.

Here is a sequence of commandds that may work (to be executed in the
directory where the repository has been cloned):

```
git checkout zhienart
git rebase trunk
git push -f abiola-zeenat zhienart
```

Basically, what should happen is that the commit that appears just after
yours when you run `git log` on your branch is the one that appears
first when you run `git log trunk`.

Abiola-Zeenat commented:
> Abiola Zeenat Abdulsalam (2022/10/28 06:40 -0700):
> Okay, I get it now, could you check if it has been updated now
> The commit message has indeed been updated, but the rebase is not visible. Here is a sequence of commandds that may work (to be executed in the directory where the repository has been cloned): ``` git checkout zhienart git rebase trunk git push -f abiola-zeenat zhienart ``` Basically, what should happen is that the commit that appears just after yours when you run `git log` on your branch is the one that appears first when you run `git log trunk`.

@shindere , I ran `git rebase Zhienart trunk` and the commit message is now visible in `git log trunk`. kindly confirm

shindere commented:
Abiola Zeenat Abdulsalam (2022/10/28 07:52 -0700):
> > Abiola Zeenat Abdulsalam (2022/10/28 06:40 -0700):
> > Okay, I get it now, could you check if it has been updated now
> > The commit message has indeed been updated, but the rebase is not visible. Here is a sequence of commandds that may work (to be executed in the directory where the repository has been cloned): ``` git checkout zhienart git rebase trunk git push -f abiola-zeenat zhienart ``` Basically, what should happen is that the commit that appears just after yours when you run `git log` on your branch is the one that appears first when you run `git log trunk`.
> 
> The rebase is now visible. kindly confirm

Yes, indeed, the PR looks ready to be merged so I am going to merge it
right now.

Many thanks for your contribution.
